qrcodereaderview/src/main/java/com/google/zxing/client/android/camera/open/OpenCameraInterface.java

```diff
@@ -40,11 +41,11 @@ private OpenCameraInterface() {
    *  camera is returned if possible or else any camera
    * @return handle to {@link OpenCamera} that was opened
    */
-  public static OpenCamera open(int cameraId) {
+  public static OpenCamera open(int cameraId, SimpleLog logger) {
```

Comment:
To be honest, I'm not very fan of passing a logger object in this kind of static methods. What do you think if instead you make `SimpleLog` final and all the methods `public static` so you can call  `SimpleLog.i("No cameras!")` for example without an instance? 
For a simple configuration inspiration you can take a look at this projects:
https://github.com/orhanobut/logger
https://github.com/JakeWharton/timber

What do you think?

Reply:
Generally I agree with you, making static methods is cleaner solution. But I also thought about using this library. This can be a very small part of the application, which is used very rarely and I did not want to leave "traces" after use, even such symbolic ones. If this were an entire application, I would have done static methods and had not thought about it.

Reply:
But then, as the library consumer, you need to always keep a reference to `SimpleLog logger` in order to call those static methods like `open`, and I don't agree with that. 
Instead, as said, maybe you can replace everything with static calls and rely on the setLoggingEnabled() method you created to log or not.